armi/reactor/converters/uniformMesh.py

```diff
@@ -85,6 +86,317 @@ def converterFactory(globalFluxOptions):
+        averageMesh = average1DWithinTolerance(numpy.array(allMeshes))
+        self._commonMesh = numpy.array(
+            [round(z, self.FLOAT_ROUNDING_DECIMALS) for z in averageMesh]
+        )
+
+    def _decuspAxialMesh(self):
+        """
+        Preserve control rod material boundaries to reduce control rod cusping effect.
+
+        Notes
+        -----
+        Uniform mesh conversion can lead to axial smearing of control assembly material, which causes
+        a pronounced control rod "cusping" affect in the differential rod worth. This function
+        modifies the uniform mesh to honor fuel and control rod material boundaries while avoiding excessively
+        small mesh sizes.
+
+        If adding control rod material boundaries to the mesh creates excessively small mesh regions,
+        this function will move internal fuel region boundaries to make room for the control rod boundaries.
+
+        This function operates by filtering out mesh points that are too close together while always holding on
+        to the specified "anchor" points in the mesh. The anchor points are built up progressively as the
+        appropriate bottom and top boundaries of fuel and control assemblies are determined.
+        """
+        # filter fuel material boundaries to mininum mesh size
+        filteredBottomFuel, filteredTopFuel = self._getFilteredFuelTopAndBottom()
+        (
+            filteredBottomControl,
+            filteredTopControl,
+        ) = self._getFilteredControlTopAndBottom(filteredBottomFuel, filteredTopFuel)
+        materialBottoms = filteredBottomFuel + filteredBottomControl
+        materialTops = filteredTopFuel + filteredTopControl
+
+        # combine the bottoms and tops into one list with bottom preference
+        allMatBounds = materialBottoms + materialTops
+        materialAnchors = self._filterMesh(
+            allMatBounds,
+            self.minimumMeshSize,
+            filteredBottomFuel + filteredTopFuel,
+            preference="bottom",
+            warn=True,
+        )
+
+        runLog.extra(
+            f"Attempting to honor control and fuel material boundaries in uniform mesh "
+            f"for {self} while also keeping minimum mesh size of {self.minimumMeshSize}. "
+            f"Material boundaries are: {allMatBounds}"
+        )
+
+        # combine material bottom boundaries with full mesh using bottom preference
+        meshWithBottoms = self._filterMesh(
+            list(self._commonMesh) + materialBottoms,
+            self.minimumMeshSize,
+            materialBottoms,
+            preference="bottom",
+        )
+        # combine material top boundaries with full mesh using top preference
+        meshWithTops = self._filterMesh(
+            list(self._commonMesh) + materialTops,
+            self.minimumMeshSize,
+            materialTops,
+            preference="top",
+        )
+        # combine all mesh points using all material boundaries as anchors with top preference
```

Comment:
it seems like this step top vs bottom is somewhat arbitrary. is that true? if so say that its arbitrary. if not explain why its not? maybe that you know 0.0 is already in there so you want 0.0 + tops

Reply:
It is arbitrary. I've gone back and forth on bottom vs. top preference.

The fuel and control material boundaries are all `anchorPoints` by now, so this is more about adjusting the mesh points away from boundaries if they are smaller than the `minimumMeshSize`. In most cases, I would expect that this filtering step doesn't actually filter out any mesh points, but there could be a situation where it becomes necessary. It's hard to say whether `bottom` or `top` is better in general here; it would depend on the specific circumstances of a given case.

Reply:
sorry, i meant to say add a comment in the code that it is arbitrary if its arbitrary. helps next person to figure out if they are breaking past intent